### ISSUE TITLE
Choice Pending in CC

### ISSUE THREAD

**Issue by GYCicada** [BG3-Community-Library-Team/BG3-Compatibility-Framework#77]
This is a very odd issue. after I switched to CF for 2e spells, it seems to have bugged out my magus mod. I was unable to go through with any level ups or CC as it warns me that there is a choice pending. the 2e spells mod I made adds spells to both Artificer and Magus spell lists through CF and artificer has no problem going through CC or level ups. When i either remove 2e spells or CF the issue is fixed.

Made sure there are no missing spell lists and no other mods uses CF in my load order. I was wondering if CF could affect the progression of a class mod that doesn't use CF by accident.

**Comment by Mythalar** [comment]
For the record we did multiple tries together and i did not encounter this issue at all.

On the other hand I could reproduce a loading stuck at 79% with only 2e spells, artificer, magus and compatibility framework mods loaded.
Does not stuck with my 150 mod list loaded :p

**Comment by Mythalar** [comment]
Regarding the lock up issue, and based on what was answered to [GYCicada](https://github.com/GYCicada), I could do some tries with this load order : Vlads Grimoire, 2e Spells, Magus, Artificer, and Compatibility Framework

If i delete config file from 2e Spells and add config to Magus/Artificer for their own spell lists the game loads indeed.

What is surprising is that there was a lock up with just those mods loaded but not with full mod list (with same relative order), so there must be something that allow bypassing the issue?

On the other hand, in my mod, I created some spells that I inserted with config file in the same mod with no issue, really weird...

**Comment by Mythalar** [comment]
Well, something I should have tried from the start :  Vlads Grimoire, 2e Spells, and Compatibility Framework only and deleted reference to modded spell lists so only vanilla classes -> lock up (expected if the spells are not there at StatsLoaded).

But full mod list it loads without console error, and spells are indeed inserted correctly.
It bothers me to no end since 2e spells definitions are only in this mod -_-

(perhaps that's because I also insert some of those new spells into other modded spell lists through a mod that is loaded after? But I did not copy/past the vanilla inserts...)

**Comment by NellsRelo** [comment]
Mentioned on Discord, but crossposting here for posterity, with regards to the lock-up issue:

With some further looking into things, I think the locking-up issue that's been mentioned has to do with when CF is loading the config files.

A few versions ago, it was loading on SessionLoad - that is, when a game session begins, either via new game or loading a save. Recently, I've switched it to load on StatsLoaded - when stats are initially set up, prior to hitting the Main Menu.

Having the config files load on SessionLoaded prevents some changes to Progressions to really take effect -namely when you add Boosts or Passives to a progression. Loading on StatsLoaded fixes this, but introduces a new, unforseen issue: if a mod introduces new spells and applies them to spell lists, the new spells don't exist at the point of insertion, causing the game to lock up.

I have a fix in the works for this, but a workaround until then would be to have a secondary patch mod that includes the config file, and must load after the mod introducing the spells

**Comment by NellsRelo** [comment]
Actually what I've written above may be incorrect, testing a local change to CF that accounts for this is still locking up, and other mods have succesfully loaded spells that they define in (Valkrana's for example). This going to be a fun one to troubleshoot 😂

**Comment by Mythalar** [comment]
For the record I removed config file from 2e spells mod, removed all reference to modded spell lists, and created a new separate mod that includes only Mods/2eSpellsLists/ScriptExtender+Meta/CompatibilityFrameworkConfig

I was full of hope for it to be a temporary workaround, it was an utter failure :D

**Comment by NellsRelo** [comment]
Some headway on the lockup issue - it was getting stuck on some duplicate values in the "Add All 1st level spells to 3rd level Magical Secret" entry. After removing them. it's still getting stuck, but at a later point (87%)

**Comment by Mythalar** [comment]
Removed those 3 item duplicate, there also is a UUID target duplicate that I removed but still stuck at 79 :p

**Comment by Mythalar** [comment]
Ok so it did launch when I removed from config file of my separate mod all references to new spell lists included in PF2e mod (IE lvl 7,8,9 full casters, 4/5 semi casters...)

**Comment by Mythalar** [comment]
Still puzzling since it not that it locks up when referencing self-introduced spell lists : 5e spells implements lvl7 spell lists so it would be ok for those, but it does not implement lvl 8 and 9 anymore, and yet game launches with when all mods are there

**Comment by Mythalar** [comment]
Found it !

It launches with full mod list because I (probably @GYCicada also) have the UnlockLevelCurve_Patch_5eSpells mod, that does reference those modded hig level spell lists.
When I add just it to the mod order CF can process the insertion.

Interesting because the actual spell lists are only defined inside PF2E mod, but UnlockLevelCurve_Patch_5eSpells mod references them in its progression file and seems it's enough.

VladGrimoire
2e spells + self defined high level lists
UnlockLevelCurve_Patch_5eSpells mod
2e spells lists (just the config file, included high level lists)
Compatibility Framework

Without UnlockLevelCurve_Patch_5eSpells mod i have to remove reference to those high level lists in config file for launch

**Comment by NellsRelo** [comment]
Aha, good catch! So ULC would need to load higher than 2eSpells to prevent the lockup?

**Comment by Mythalar** [comment]
I've done so many tries/changes that I cannot reproduced it -_-'

I could at least reproduced the game launch by removing all insertion into non vanilla spell lists, so it seems issus is with spell lists themselves, that would be why Valdra is not affected, IIRC it does not create new lists?

**Comment by Mythalar** [comment]
O ki've done things proper and am quite sure the spell lists are the issue : 
Vlad + 2e spells + CF = lock up
Vlad + 2e spells (minus lists folder) + CF = launch
Vlad + magus + artificer + 2e spells + CF = lock up
Vlad + magus + artificer +2e spells (minus lists folder) + CF = launch

there are ytarghets errors because there must be some mistakes, but it launches without the new lists inside 2e spells

**Comment by Mythalar** [comment]
Oh it was actually simple : throw 5e spells mod in the mod list, either before or after 2e spells, (unmodified) and it launches...

**Comment by GYCicada** [comment]
that seems very odd to me. I made the progressions folder because 5e spells and ULC removed the 8th and 9th level spell lists from their mods, so i re-added them back in with 2e spells. hmmm.

**Comment by Mythalar** [comment]
Going back to this : 
> Having the config files load on SessionLoaded prevents some changes to Progressions to really take effect -namely when you add Boosts or Passives to a progression. Loading on StatsLoaded fixes this, but introduces a new, unforseen issue: if a mod introduces new spells and applies them to spell lists, the new spells don't exist at the point of insertion, causing the game to lock up.

Perhaps this issue is a concern for spell lists and not spells, so some other mod has to implement the spell lists?

But would still be odd since you reimplemented some spell lists that 5e mod author removed (mainly arcane caster 8 and 9 level spells)... Looks like CF needs a headstrat and then it can proceed (or 8 and 9 level spells do not actually work and we cannot see it since you don't insert them yet?)

**Comment by GYCicada** [comment]
i have not added any 8th or 9th level spells yet, as i have just finished 7th.

**Comment by Mythalar** [comment]
Yep I've seen your new release, was just wondering how those lists will work, since those are the only lists with UUID that are not already in 5e spells

**Comment by GYCicada** [comment]
essentially i created those lists in the 2e spells mod and modified the corresponding progressions to use those lists when selecting / adding higher level spells.

**Comment by Mythalar** [comment]
Which is only logical, and the UUID you used for high level lists are the same that 5e spells used previously (also used by ULC)...

New CF release does not fix the issue even by switching back to using On StatsLoaded, so there must be something else.

**Comment by NellsRelo** [comment]
Just noting that I am still looking into this - haven't been able to reproduce the Choices Pending issue, but trying to find ways to cleanly handle the loading-screen locking issue
